inc/saiport.h

```diff
@@ -478,6 +486,15 @@ typedef enum _sai_port_attr_t
     SAI_PORT_ATTR_ADVERTISED_FEC_MODE,
 
     /**
+     * @brief Advertised port FEC AN request bit.  True means FEC is requested for 10/40/100G
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_PORT_ATTR_ADVERTISED_FEC_MODE_REQUEST,
```

Comment:
So in IEEE 802.3by-2016 Clause 73.6.5, there are 4 bits (F0 - F3)

1. F0 = 10 Gb/s - FEC ability
2. F1 = 10 Gb/s - FEC requested
3. F2 = 25G RS-FEC requested
4. F3 = 25G FC-FEC requested

These are the bits SAI_PORT_ATTR_ADVERTISED_FEC_MODE_* modify.  The interpretation is:

1. SAI_PORT_ATTR_ADVERTISED_FEC_MODE modifies F0, F2, F3
2. SAI_PORT_ATTR_ADVERTISED_FEC_MODE_REQUEST modifies F1

Your question is why do we need REQUEST?  Why not just have F0 & F1 always both == 0 or both == 1 (thus @validonly would make sense)?

Answer is some customers want ability == 1 and request == 0 (saying I'm capable of FEC but I don't wan't to use it).  So there is no need to use @validonly.

25G consortium has multiple bits for requesting RS vs FC vs none.  But since I don't see SAI making a distinction between consortium vs IEEE, folks we work with default to IEEE standard and switch to consortium via sai.ini file

If we want to support 25G consortium request via this new attribute, we could change it to sai_port_fec_mode_t